source/kraken/realtime.cpp

```diff
@@ -89,8 +99,11 @@ static bool check_trip_update(const transit_realtime::TripUpdate& trip_update) {
         for (const auto& st: trip_update.stop_time_update()) {
             uint32_t arrival_time = st.arrival().time();
             uint32_t departure_time = st.departure().time();
+            bool deleted = is_deleted(st.arrival()) ||
```

Comment:
We should check is_deleted(arrival) when dealing with arrival consistency, and separately is_deleted(departure) when dealing with departure consistency (so below).
And probably keep a `last_stop_event_time` to use for comparison.